apps/condo/domains/news/schema/NewsItem.test.js

```diff
@@ -287,8 +293,125 @@ describe('NewsItems', () => {
+                const unitType2 = FLAT_UNIT_TYPE
+                const unitName2 = faker.lorem.word()
+                const unitType3 = FLAT_UNIT_TYPE
+                const unitName3 = faker.lorem.word()
+
+                await createTestResident(adminClient, residentClient1.user, property, {
+                    unitType: unitType1,
+                    unitName: unitName1,
+                })
+
+                await createTestResident(adminClient, residentClient2.user, property, {
+                    unitType: unitType2,
+                    unitName: unitName2,
+                })
+
+                await createTestResident(adminClient, residentClient3.user, residentClient3.property, {
+                    unitType: unitType3,
+                    unitName: unitName3,
+                })
+
+                // News item for particular unit
+                const [newsItem1, newsItem1Attrs] = await createTestNewsItem(adminClient, o10n)
+                const [newsItemScope1] = await createTestNewsItemScope(adminClient, newsItem1, {
+                    property: { connect: { id: property.id } },
+                    unitType: unitType1,
+                    unitName: unitName1,
+                })
+
+                // News item for property
+                const [newsItem2, newsItem2Attrs] = await createTestNewsItem(adminClient, o10n)
+                const [newsItemScope2] = await createTestNewsItemScope(adminClient, newsItem2, {
+                    property: { connect: { id: property.id } },
+                })
+
+                // News item for all organization
+                const [newsItem3, newsItem3Attrs] = await createTestNewsItem(adminClient, o10n)
+                const [newsItemScope3] = await createTestNewsItemScope(adminClient, newsItem3)
+
+                const newsItems1 = await NewsItem.getAll(residentClient1, {})
+                const newsItems2 = await NewsItem.getAll(residentClient2, {})
+                const newsItems3 = await NewsItem.getAll(residentClient3, {})
+
+                expect(newsItems1).toHaveLength(3)
+                expect(newsItems1).toEqual(expect.arrayContaining([
+                    expect.objectContaining({ id: newsItem1.id, title: newsItem1Attrs.title }),
+                    expect.objectContaining({ id: newsItem2.id, title: newsItem2Attrs.title }),
+                    expect.objectContaining({ id: newsItem3.id, title: newsItem3Attrs.title }),
+                ]))
+
+                expect(newsItems2).toHaveLength(2)
+                expect(newsItems2).toEqual(expect.arrayContaining([
+                    expect.objectContaining({ id: newsItem2.id, title: newsItem2Attrs.title }),
+                    expect.objectContaining({ id: newsItem3.id, title: newsItem3Attrs.title }),
+                ]))
+
+                expect(newsItems3).toHaveLength(0)
+            })
+
+            test('user with two residents within organization must see all eligible news items', async () => {
```

Comment:
also you can check the case when the user with 2 residents in different organizations

Reply:
Donw